File: Controllers/TextController.vb
Imports System
Imports System.Web
Imports System.Web.Http
Imports System.Collections.Generic
Imports System.Linq
Imports AttributeRouting.Web.Http
Imports System.Data.SqlClient
Imports System.Web.Script.Serialization
Imports log4net
Imports System.Net
Imports Newtonsoft.Json

Namespace CalcmenuAPI
    Public Class TextController
        Inherits ApiController

        Private Shared ReadOnly Log As ILog = LogManager.GetLogger(System.Reflection.MethodBase.GetCurrentMethod().DeclaringType)


        <HttpPost> <[POST]("/api/text/search")> _
        Public Function GetTextByName2(data As Models.ConfigurationcSearch) As List(Of Models.Text)
            Try
                Dim ds As New DataSet
                Using cmd As New SqlCommand
                    With cmd
                        Using cn As New SqlConnection(ConnectionString)
                            Try
                                .Connection = cn
                                .CommandType = CommandType.StoredProcedure
                                .CommandText = "[dbo].[API_SEARCH_TEXT]"
                                .Parameters.Add("@nvcName", SqlDbType.NVarChar, 2000).Value = data.Name
                                .Parameters.Add("@intCodeTrans", SqlDbType.Int).Value = data.CodeTrans
                                .Parameters.Add("@rowsPerPage", SqlDbType.Int, 50).Value = 10
                                .Parameters.Add("@namefiltertype", SqlDbType.Int, 10).Value = 4
                                .Parameters.Add("@skip", SqlDbType.Int, 0).Value = 0

                                cn.Open()
                                Dim _da As New SqlDataAdapter(cmd)
                                _da.Fill(ds)
                            Finally
                                If Not cn Is Nothing Then
                                    cn.Close()
                                    CType(cn, IDisposable).Dispose()
                                End If
                            End Try
                        End Using
                    End With
                End Using
                Dim texts As New List(Of Models.Text)
                For Each r In ds.Tables(0).Rows
                    texts.Add(New Models.Text With {
                                    .TextCode = GetInt(r("Code")), _
                                    .TextName = r("Name"), _
                                    .TextDate = GetBool(r("Dates"))})
                Next


                If data.Name.Trim <> "" Then

                    Dim textresult As New List(Of Models.Text)

                    'Dim arrNames() As String = data.Name.Trim.Split(",")
                    'For Each word In arrNames
                    '    If word.Trim.ToString <> "" Then    'RDTC 2015.08.10; added this otherwise the rows are getting duplicated
                    '        For Each s In taxes
                    '            If s.TaxValue.ToString.ToLower.Contains(Common.ReplaceSpecialCharacters(word.Trim.ToLower)) Or _
                    '                s.TaxName.ToString.ToLower.Contains(Common.ReplaceSpecialCharacters(word.Trim.ToLower)) Then
                    '                taxresult.Add(s)
                    '            End If
                    '        Next
                    '    End If
                    'Next
                    texts = textresult

                End If

                Return texts.ToList
            Catch aex As ArgumentException
                Log.Warn(System.Reflection.MethodBase.GetCurrentMethod().Name + ": Missing or invalid parameters", aex)
                Throw New HttpResponseException(GenericErrorResponse("Request failed", HttpStatusCode.BadRequest, 440))
            Catch hex As HttpResponseException
                Log.Error(System.Reflection.MethodBase.GetCurrentMethod().Name + ": Unexpected error occured", hex)
                Throw hex
            Catch ex As Exception
                Log.Error(System.Reflection.MethodBase.GetCurrentMethod().Name + ": Unexpected error occured", ex)
                Throw New HttpResponseException(GenericErrorResponse("Request failed", HttpStatusCode.InternalServerError, 500))
            End Try
        End Function


    End Class
End Namespace

File: src/CmcStandardPrinting.Api/Controllers/TextController.cs
using System;
using System.Collections.Generic;
using System.Data;
<<<<<<< HEAD
using Microsoft.Data.SqlClient;
=======
using System.Data.SqlClient;
>>>>>>> main
using CmcStandardPrinting.Domain.Common;
using CmcStandardPrinting.Domain.Texts;
using Microsoft.AspNetCore.Mvc;
using Microsoft.Extensions.Configuration;
using Microsoft.Extensions.Logging;

namespace CmcStandardPrinting.Api.Controllers;

[ApiController]
[Route("api/[controller]")]
public class TextController : ControllerBase
{
    private readonly IConfiguration _configuration;
    private readonly ILogger<TextController> _logger;

    public TextController(IConfiguration configuration, ILogger<TextController> logger)
    {
        _configuration = configuration;
        _logger = logger;
    }

    private string ConnectionString => _configuration.GetConnectionString("Default") ?? string.Empty;

    [HttpPost("search")]
    public ActionResult<List<Text>> GetTextByName2([FromBody] ConfigurationcSearch data)
    {
        try
        {
            var ds = new DataSet();
            using var cmd = new SqlCommand();
            using var cn = new SqlConnection(ConnectionString);
            cmd.Connection = cn;
            cmd.CommandType = CommandType.StoredProcedure;
            cmd.CommandText = "[dbo].[API_SEARCH_TEXT]";
            cmd.Parameters.Add("@nvcName", SqlDbType.NVarChar, 2000).Value = data.Name ?? string.Empty;
            cmd.Parameters.Add("@intCodeTrans", SqlDbType.Int).Value = data.CodeTrans;
            cmd.Parameters.Add("@rowsPerPage", SqlDbType.Int, 50).Value = 10;
            cmd.Parameters.Add("@namefiltertype", SqlDbType.Int, 10).Value = 4;
            cmd.Parameters.Add("@skip", SqlDbType.Int, 0).Value = 0;

            cn.Open();
            using var da = new SqlDataAdapter(cmd);
            da.Fill(ds);

            var texts = new List<Text>();
            foreach (DataRow r in ds.Tables[0].Rows)
            {
                texts.Add(new Text
                {
                    TextCode = GetInt(r["Code"]),
                    TextName = GetStr(r["Name"]),
                    TextDate = GetBool(r["Dates"]).ToString(),
                });
            }

            if (!string.IsNullOrWhiteSpace(data.Name))
            {
                // Legacy behavior returns an empty list when a name filter is provided.
                texts = new List<Text>();
            }

            return Ok(texts);
        }
        catch (ArgumentException ex)
        {
            _logger.LogWarning(ex, "Missing or invalid parameters for text search");
            return BadRequest();
        }
        catch (SqlException ex)
        {
            _logger.LogError(ex, "Database error during text search");
            return StatusCode(500);
        }
        catch (Exception ex)
        {
            _logger.LogError(ex, "Unexpected error during text search");
            return StatusCode(500);
        }
    }

    private static int GetInt(object? value, int fallback = 0)
    {
        if (value == null || value == DBNull.Value) return fallback;
        return Convert.ToInt32(value);
    }

    private static bool GetBool(object? value)
    {
        if (value == null || value == DBNull.Value) return false;
        return Convert.ToBoolean(value);
    }

    private static string GetStr(object? value)
    {
        if (value == null || value == DBNull.Value) return string.Empty;
        return value.ToString() ?? string.Empty;
    }
}
